### ISSUE TITLE
Regression Tests Failures

### ISSUE THREAD

**Issue by domainexpert** [tracer-x/klee#77]
Need to ensure that all regression test suites succeeds. This should be done before resolving #20  so as merging can be done carefully commit by commit with proper regression testing done at each step.

**Comment by domainexpert** [comment]
@feliciahalim As at  7a0f432 `test/Feature/MemoryLimit.c` not only fails, it slows down the regression tests by about 10 minutes on my machine, which is not normal.

**Comment by domainexpert** [comment]
@feliciahalim The above was run by also deleting `test/Runtime`. Deleting only `test/Feature/MemoryLimit.c` still results in the same slowdown, so there is at least another culprit in `test/Runtime`. But at least we know one of the problem test is `test/Feature/MemoryLimit.c` so hopefully we can start fixing the issue.

**Comment by domainexpert** [comment]
@feliciahalim The problem is that `ITreeNode::markPathCondition` is really slow. Aborting the KLEE execution of `MemoryLimit.c` reveals the following statistics:
```
KLEE: output directory is "/home/dcsandr/software/klee/test/Feature/MemoryLimit.klee-out"
Starting Z3Solver ...
KLEE: WARNING: undefined reference to function: printf
KLEE: WARNING ONCE: calling external: printf(23323824)
KLEE: ctrl-c detected, requesting interpreter to halt.
KLEE: halting execution, dumping remaining states

KLEE: done: Subsumption statistics
KLEE: done:     Time for actual solver calls in subsumption check (ms) = 0.000000e+00
KLEE: done:     Number of solver calls for subsumption check (failed) = 0 (0)
KLEE: done:     Average table entries per subsumption checkpoint = 1.00
KLEE: done:     Average solver calls per subsumption check = 0.00
KLEE: done: ITree method execution times (ms):
KLEE: done:     setCurrentINode = 3.550000e+00
KLEE: done:     remove = 2.340000e+00
KLEE: done:     subsumptionCheckTimer = 0.000000e+00
KLEE: done:     markPathCondition = 7.721758e+04
KLEE: done:     split = 0.000000e+00
KLEE: done:     executeOnNode = 1.361501e+05
KLEE: done: ITreeNode method execution times (ms):
KLEE: done:     getInterpolant = 0.000000e+00
KLEE: done:     addConstraintTime = 0.000000e+00
KLEE: done:     splitTime = 0.000000e+00
KLEE: done:     execute = 1.361440e+05
KLEE: done:     bindCallArguments = 0.000000e+00
KLEE: done:     bindReturnValue = 0.000000e+00
KLEE: done:     getStoredExpressions = 0.000000e+00
KLEE: done:     getStoredCoreExpressions = 0.000000e+00
KLEE: done:     computeCoreAllocations = 1.068000e+00

KLEE: done: total instructions = 16614
KLEE: done: completed paths = 1, among which
KLEE: done:     early-terminating paths (instruction time limit, solver timeout, max-depth reached) = 1
KLEE: done:     subsumed paths = 0
KLEE: done:     error paths = 0
KLEE: done:     program exit paths = 0
KLEE: done: generated tests = 1, among which
KLEE: done:     early-terminating tests (instruction time limit, solver timeout, max-depth reached) = 1
KLEE: done:     subsumed tests = 0
KLEE: done:     error tests = 0
KLEE: done:     program exit tests = 0
```
I think for this particular problem it would be best to first resolve #85, while keeping in mind #101 .

**Comment by domainexpert** [comment]
@feliciahalim Other failed regression tests as at 7a0f432 are:
```
Runtime/POSIX/DirConsistency.c
Runtime/POSIX/DirSeek.c
Runtime/POSIX/Write2.c
```
They are all long running, so presumably they suffer from the same problem as `MemoryLimit.c`. For these, I recommend resolving #85 first.

**Comment by domainexpert** [comment]
Current regression tests by default run with STP core solver. We need to enable another set of run which uses Z3 + lazy annotation.

**Comment by domainexpert** [comment]
In order to run the regression tests with Z3 and interpolation, one could add modify the following lines in `test/Makefile`:
```
ifdef VERBOSE
LIT_ARGS := -v
else
LIT_ARGS := -s -v
endif
```
into
```
ifdef VERBOSE
LIT_ARGS := -v --param klee_opts="-solver-backend=z3"
else
LIT_ARGS := -s -v --param klee_opts="-solver-backend=z3"
endif
```

**Comment by domainexpert** [comment]
Resolved via #165.
